add ability to build from local PKG-INFO file

huakim opened the pull request.

mcepl reviewed (changes requested):
Before I will be even looking at diff, could I ask for `git rebase -i` and rearranging commits to some meaningful units with proper commit messages?

https://who-t.blogspot.com/2009/12/on-commit-messages.html

huakim commented:
> Before I will be even looking at diff, could I ask for `git rebase -i` and rearranging commits to some meaningful units with proper commit messages?
> 
> https://who-t.blogspot.com/2009/12/on-commit-messages.html

I can, but lasy, code just work, and there isn't much changes done, just added two functions 'pypi_text_file' and 'pypi_json_file', also added --local, --localfile and --source-extension for generate subcommand. The --local flag indicates, that py2pack should fetch data from local file such as '{name}.egg-info/PKG-INFO' by default, --localfile indicates direct path to the local file. --source-extension is extension for source, if no source url is provided, by default it takes value '.zip'.  Also, i merged commits from danigm user (https://github.com/openSUSE/py2pack/pull/178), who did patch for using build.project_wheel_metadata to extract more metadata when it's not using setuptools and the metaextract package is not useful. There changes provided as is, with no any responsibilities, just for meself

mcepl commented:
>  just for meself

If you made changes just for yourself, then I don’t know why you submitted it here.
